### PR TITLE
Update natural language parser

### PR DESCRIPTION
**Pull request by ColeDCrawford**
This PR updates the natural language parser to work with the 2018 spec. As noted in the [EDTF docs](https://www.loc.gov/standards/datetime/), 

> This specification differs from the earlier draft as follows:
> 
> - the unspecified date character (formerly lower case ‘u’) is superseded by the character (upper case) 'X';
> - Masked precision is eliminated;
> - the uncertain and approximate qualifiers, '?' and '~',  when applied together, are combined into a single qualifier character '%';
> - “qualification from the left” is introduced and replaces the grouping mechanism using parentheses;
> - the extended interval syntax  keywords 'unknown' and 'open' have been replaced with null and the double-dot notation ['..'] respectively;
> - the year prefix 'y' and the exponential indicator 'e', both previously lowercase, are now 'Y' and 'E' (uppercase); and
> - the significant digit indicator 'p' is now 'S' (uppercase).

Checklist for the PR:

- [x] Replace the unspecified date character `u` with `X`
- [x] Remove masked precision
- [x] Combine `?~` into `%` for uncertain + approximate
- [x] Use qualification from the left rather than parentheses for grouping
- [x] Replace `unknown` with `null` syntax in intervals
- [x] Replace `open` with `..` syntax syntax in intervals
- [x] Replace `y`, `e`, and `p` with `Y`, `E`, and `S` syntax for exponential years and significant digits

### CODE REVIEW

**Comment by ColeDCrawford** [comment]
@aweakley do you want to remove the old parentheses grouping examples from `BAD_EXAMPLES` here? Those are from the old spec.

I don't see any code for the old `open` syntax for intervals?

**Comment by ColeDCrawford** [comment]
Not sure why the tests for this are permanently queued, but clicking into one shows that they have succeeded

**Comment by aweakley** [comment]
Do the parentheses ones make things more complex? I quite like that it's clear from the tests that inputs like that will raise an error, but if it'll make our lives easier to get rid of them then I'm happy with that.

File: edtf/parser/tests.py [hunk 1]
```diff
@@ -111,29 +111,35 @@
     ('2010-24', ('2010-12-01', '2010-12-31')),
 
     # ******************************* LEVEL 2 *********************************
-    # Partial Uncertain/Approximate
+    # Qualification
+    # Group qualification: a qualification character to the immediate right of a component applies
+    # to that component as well as to all components to the left.
+    # year, month, and day are uncertain and approximate
+    ('2004-06-11%', ('2004-06-11', '2004-06-09', '2004-06-13')),
```
Comment (aweakley) [inline edtf/parser/tests.py:118]: Shouldn't this one be a broader range, given the year and month are also uncertain and approximate?

**Comment by ColeDCrawford** [comment]
> Do the parentheses ones make things more complex? I quite like that it's clear from the tests that inputs like that will raise an error, but if it'll make our lives easier to get rid of them then I'm happy with that.

Nope, they fail just fine as they currently are so I'm also happy to leave them.

File: edtf/parser/tests.py (hunk 1 above)
Comment (aweakley) [reply edtf/parser/tests.py:118]: You can say `2002%-06-11` instead if you want the narrow range.  So I think we should fuzz all parts to the left of the % symbol, and we'd get the broader "2002-04-09" - "2006-08-13" range.

**Comment by ColeDCrawford** [comment]
Not sure why the tests are hanging but they look like they are passing if you click in.

I've updated the tests, specifically EDTF level 1 "Qualification of a date (complete)" dates. These parse as `UncertainOrApproximate` classes.

**Comment by aweakley** [comment]
This looks great, thank you.

**Pull request merged**